tests/chains/ethereum/scripts/cmd/helper/chain.go

```diff
@@ -0,0 +1,258 @@
+package helper
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	gethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/client"
+	ibccommitment "github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/ibccommitmenttesthelper"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/ibchandler"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/ics20bank"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/ics20transferbank"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/contract/simpletoken"
+	ibcclient "github.com/hyperledger-labs/yui-ibc-solidity/pkg/ibc/core/client"
+	"github.com/hyperledger-labs/yui-ibc-solidity/pkg/wallet"
+)
+
+type Chain struct {
+	chainID        int64
+	client         *client.ETHClient
+	lc             *LightClient
+	mnemonicPhrase string
+	keys           map[uint32]*ecdsa.PrivateKey
+
+	ContractConfig ContractConfig
+
+	// Core Modules
+	IBCHandler    ibchandler.Ibchandler
+	IBCCommitment ibccommitment.Ibccommitmenttesthelper
+
+	// App Modules
+	SimpleToken   simpletoken.Simpletoken
+	ICS20Transfer ics20transferbank.Ics20transferbank
+	ICS20Bank     ics20bank.Ics20bank
+
+	// State
+	LastLCState LightClientState
+
+	// IBC specific helpers
+	ClientIDs   []string      // ClientID's used on this chain
+	Connections []*Connection // track connectionID's created for this chain
+	IBCID       uint64
+
+	// Channel specific helpers
+	Channel Channel
+}
+
+func NewChain(chainInfo ChainInfo, client *client.ETHClient, lc *LightClient, config ContractConfig, mnemonicPhrase string, ibcID uint64) *Chain {
+	ibcHandler, err := ibchandler.NewIbchandler(config.GetIBCHandlerAddress(), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	ibcCommitment, err := ibccommitment.NewIbccommitmenttesthelper(config.GetIBCCommitmentTestHelperAddress(), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	simpletoken, err := simpletoken.NewSimpletoken(config.GetSimpleTokenAddress(), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	ics20transfer, err := ics20transferbank.NewIcs20transferbank(config.GetICS20TransferBankAddress(), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	ics20bank, err := ics20bank.NewIcs20bank(config.GetICS20BankAddress(), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+
+	return &Chain{
+		client:         client,
+		chainID:        int64(chainInfo.ETHChainID),
+		lc:             lc,
+		ContractConfig: config,
+		mnemonicPhrase: mnemonicPhrase,
+		keys:           make(map[uint32]*ecdsa.PrivateKey),
+		IBCID:          ibcID,
+
+		IBCHandler:    *ibcHandler,
+		IBCCommitment: *ibcCommitment,
```

Comment:
Do the CLI tools use this `IBCCommitment` somewhere?

Please delete any code that has been copy-pasted from contract_test.go and is not needed for these tools.

Reply:
Removed unused code